### PR TITLE
Add Yourls

### PR DESCRIPTION
**Pull request by steigerdich**
First pull request... :)

- [x] I have tested the template using the method described in README.md thoroughly
- [x] I have ensured that I put as much default values as possible (except passwords) to ensure minimum effort required for end users to get started.
- [x] I have ensured that I am not using the "latest" tag as this tag is dynamically changing and might break the one-click app. Use a fixed version.
- [x] I have made sure that instructions.start and instructions.end are clear and self-explanatory.
- [x] Icon is added as a png file to the logos directory.
- [x] I've executed the checks if necessary by running `npm ci && npm run validate_apps && npm run formatter`
- [x] I will take responsibility addressing any issues that arises as a result of this PR (maintaining this app).

After I had problems setting up the URL shortener Kutt I made a YAML for YOURLS.

### CODE REVIEW

File: public/v4/apps/yourls.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            MYSQL_USER: $$cap_db_user
+            MYSQL_PASSWORD: $$cap_db_pass
+        caproverExtra:
+            notExposeAsWebApp: 'true'
+    $$cap_appname:
+        depends_on:
+            - $$cap_appname-db
+        image: yourls:$$cap_yourls_version
+        volumes:
+            - $$cap_appname-logs:/usr/src/yourls/logs
```
Comment (githubsaturn) [inline public/v4/apps/yourls.yml:20]: Not familiar with yourls - but is mapping the logs as a persistent volume really necessary?

File: public/v4/apps/yourls.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    instructions:
+        start: >-
+            YOURLS is a set of PHP scripts that will allow you to run Your Own URL Shortener.
+        end: >
+            yourls is ready for final steps at $$cap_YOURLS_SITE/admin .
+            In case you have set another Instance URL than http://$$cap_appname.$$cap_root_domain: Don't forget to connect! (and enable HTTPS)
+
+            Have fun!
+    displayName: yourls
+    isOfficial: false
```
Comment (githubsaturn) [inline public/v4/apps/yourls.yml:80]: I think you're using the official image.

**Review by githubsaturn — COMMENTED** [review]
Thanks! 

Just a few minor comments/questions. Also, a few checks from the self checkbox list are missing.

File: public/v4/apps/yourls.yml (hunk 1 above)
Comment (steigerdich) [reply public/v4/apps/yourls.yml]: You are right, not necessary!

File: public/v4/apps/yourls.yml (hunk 2 above)
Comment (steigerdich) [reply public/v4/apps/yourls.yml]: True! Changed...

**Comment by steigerdich** [comment]
I will make a few more tests tomorrow, maybe it would be nice to publish another version that already comes with a frontend. I'll check that..

**Comment by githubsaturn** [comment]
Thanks! Can you also please check all the self check list in the PR?

**Comment by steigerdich** [comment]
Small proof that it already works: https://tinyy.de/pryourls ;)

**Comment by githubsaturn** [comment]
Fantastic! Thanks!

PS: no need for a proof, just a self check suffices :smile:

**Comment by ms4sman** [comment]
I may have discovered an issue with this one-click app. The cap_YOURLS_COOKIEKEY doesn't seem to get used anywhere, and possibly other variables. I do not know all the impacts of this, but what I have found is that since the value I supply to that variable is not being used, the Secret Signature Token is always the same in every install I do, which is obviously a security issue. That CookieKey is used to generate said signature token.

Let me know if more information is needed!

Thanks

**Pull request merged**